runtime/include/cloe/databroker.hpp

```diff
@@ -0,0 +1,1254 @@
+ */
+template<typename TValue>
+constexpr bool is_wanted_type = std::is_same<TValue, typename std::remove_cv<typename std::remove_pointer<typename std::remove_reference<TValue>::type>::type>::type>::value;
+/**
+ * SFINAE template to determine return-type for enabling functions consuming wanted type T
+ */
+template<typename TValue, typename TReturn>
+using enable_if_is_wanted = std::enable_if<is_wanted_type<TValue>, TReturn >;
+
+#define __HAVE_BETTER_ERROR_MESSAGES 1
+
+#if defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+
+/**
+ * SFINAE template to determine return-type for enabling functions consuming unwanted type T
+ */
+template<typename TValue, typename TReturn>
+using enable_if_is_unwanted = std::enable_if<!is_wanted_type<TValue>, TReturn >;
+/**
+ * Constexpr which evaluates to false while template expansion phase
+ * \note: Basic idea is to enforce compiler errors only when the template is expanded.
+ */
+template <typename>
+constexpr auto always_false = false;
+
+/**
+ * @brief Causes an understandable compiler error, instead of cumbersome template errors in case unsuitable types are used on the API
+ * \note: Basic idea is to enforce compiler errors only when the template is expanded.
+ */
+template<typename TValue>
+constexpr void cause_compiler_error_due_to_unwanted_type() { static_assert(always_false<TValue>, "Please call the method with an unqualified base type (not pointer, reference, const, volatile). Please find the offending LOC in above line 'require from here'."); }
+
+#endif // defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+
+/**
+ * Abstract event implementation
+ * @tparam TArgs Arguments of the eventhandler function
+ * @note: Design-Goals:
+ * - Design-#1: Unsubscribing from an event is not intended
+ */
+template<typename... TArgs>
+class Event
+{
+private:
+    using eventhandler = std::function<void(TArgs...)>;
+    using eventhandler_collection = std::vector<eventhandler>;
+    using size_type = std::size_t;
+    eventhandler_collection eventhandlers_{};
+public:
+    /**
+     * @brief Adds an eventhandler to this event
+     * @param eventhandler 
+     */
+    void add(eventhandler eventhandler)
+    {
+        eventhandlers_.push_back(std::move(eventhandler));
+    }
+    size_type count() const
+    {
+        return eventhandlers_.size();
+    }
+    /**
+     * @brief Raises this event
+     * @param args Parameters of this event
+     */
+    void raise(TArgs... args) const
+    {
+        for (const auto& eventhandler : eventhandlers_)
+        {
+            try
+            {
+                eventhandler(std::forward<decltype(args)>(args)...);
+            }
+            catch (...)
+            {
+                throw;
+            }
+        }
+    }
+};
+
+// Forward declarations
+template <typename TValue>
+class Container;
+class Signal;
+class DataBroker;
+
+template <typename TValue>
+class Container

```

Comment:
The name `Container` feels so... generic, that I feel it is somewhat misleading.
In essence, it exists for notification purposes, right?

What do you think about something like `Notifier` as a class name?